회원탈퇴 기능 개발

Habist opened the pull request:
## 작업 내용
회원탈퇴 시 회원정보 삭제와 인증된 로그인 정보를 삭제하는 형태로 구현하였습니다.

>  참조이슈
> #15

Habist committed:
feat: [#15] 회원정보 삭제 기능 개발

회원정보 삭제 기능을 추가하였습니다.

f-lab-dev commented on src/main/java/com/campool/controller/UserController.java:
```diff
@@ -46,4 +47,11 @@ public void updateUser(@Valid UserUpdateRequest userUpdateRequest, @PresentUserI
 
+    @LoginValidation
+    @DeleteMapping("/users")
+    public void deleteUser(@PresentUserId String id) {
```
이름을 좀 더 보니 `@PresentUserId`보다는 `@LoginUserId` 혹은 `@CurrentUserId` 이런식으로 표현하는게 더 명확해보이는데 어떠실까요?

Habist replied:
**Current는 존재 자체의 뉘앙스가 강하고 Present는 지금 이 순간이라는 뉘앙스가가 강하다고 알려져 있습니다.**

처음에는 로그인 요청을 기준으로 요청하는 **순간**에 유저 ID를 표현하고 싶었는데 다시 보니 요청보다는 ID를 기준으로 **존재**하는 유저 ID가 좀 더 괜찮을 것 같다는 생각이 드네요! Current도 좋지만 Login이라는 명확한 개념이 더 좋을 것 같아서 `LoginUserId`로 변경하였습니다!

Habist committed:
refactor: [#15 - 리뷰 반영] PresentUserId 명칭 및 현재 비밀번호 변수 명 변경

현재 로그인된 사용자 ID의 의미를 좀 더 명확하게 표현할 수 있는 단어 Login으로 변경하였습니다.
현재 비밀번호라는 의미의 변수 명을 present에서 존재 의미가 강한 단어인 current로 변경하였습니다.
